packages/git-diff/lib/main.js

```diff
@@ -1,32 +1,52 @@
-const GitDiffView = require('./git-diff-view');
-const DiffListView = require('./diff-list-view');
+'use babel';
+
+import { CompositeDisposable } from 'atom';
+import GitDiffView from './git-diff-view';
+import DiffListView from './diff-list-view';
 
 let diffListView = null;
+let diffViews = null;
+let subscriptions = null;
+
+export default {
+  activate(state) {
+    subscriptions = new CompositeDisposable();
+    diffViews = new Set();
+
+    subscriptions.add(
+      atom.workspace.observeTextEditors(editor => {
+        const editorElement = atom.views.getView(editor);
+        const diffView = new GitDiffView(editor);
 
-module.exports = {
-  activate() {
-    const watchedEditors = new WeakSet();
-
-    atom.workspace.observeTextEditors(editor => {
-      if (watchedEditors.has(editor)) return;
-
-      new GitDiffView(editor).start();
-      atom.commands.add(
-        atom.views.getView(editor),
-        'git-diff:toggle-diff-list',
-        () => {
-          if (diffListView == null) diffListView = new DiffListView();
-          diffListView.toggle();
-        }
-      );
-
-      watchedEditors.add(editor);
-      editor.onDidDestroy(() => watchedEditors.delete(editor));
-    });
+        diffViews.add(diffView);
+
+        let editorSubs;
+        const command = 'git-diff:toggle-diff-list';
+        subscriptions.add(
+          (editorSubs = new CompositeDisposable(
+            atom.commands.add(editorElement, command, () => {
+              if (diffListView == null) diffListView = new DiffListView();
+              diffListView.toggle();
+            }),
+            editor.onDidDestroy(() => {
+              diffView.destroy();
+              diffViews.delete(diffView);
+              editorSubs.dispose();
+              subscriptions.remove(editorSubs);
+            })
+          ))
+        );
+      })
+    );
   },
 
   deactivate() {
-    if (diffListView) diffListView.destroy();
     diffListView = null;
+
+    for (const diffView of diffViews) diffView.destroy();
```

Comment:
Please use `{}`. This code is not debuggable and hard to see the sentence

Reply:
Thanks for making me aware of how incomplete the chromium V8 JavaScript Debugging is lol. At first I didn't believe you because I've never had to debug a compound statement in JS before. Decided to try it out. Sucks. I'll fix those.

Reply:
I can usually set breakpoints anywhere within a line of code. On any method, usually. So I would be able to set one on `.destroy()` I think, if that's what you mean @aminya.

Using `chrome://inspect` in Google Chrome of macOS 10.15.7.

Reply:
Even if you can debug this inside the latest Chrome, you can't do it inside Atom. 

This is a not-so-good style of coding. There are other things that are wrong with them.

See this video from Tony Van Erd to learn more:
https://youtu.be/GPP64opjy_Y?t=1068

Reply:
> Even if you can debug this inside the latest Chrome, you can't do it inside Atom.

For what it's worth: We do continue to get newer Chrome via newer Electron. That's one good reason to finish work on the Electron 9 upgrade (https://github.com/atom/atom/pull/21777).